src/iperf_util.c

```diff
@@ -433,56 +434,65 @@ iperf_dump_fdset(FILE *fp, char *str, int nfds, fd_set *fds)
     fprintf(fp, "]\n");
 }
 
-/* daemon(3) implementation for systems lacking one */
+/*
+ * daemon(3) implementation for systems lacking one.
+ * Cobbled together from various daemon(3) implementations,
+ * not intended to be general-purpose. */
 #ifndef HAVE_DAEMON
 int daemon(int nochdir, int noclose)
 {
     pid_t pid = 0;
     pid_t sid = 0;
+    int fd;
+
+    /*
+     * Ignore any possible SIGHUP when the parent process exits.
+     * Note that the iperf3 server process will eventually install
+     * its own signal handler for SIGHUP, so we can be a little
+     * sloppy about not restoring the prior value.  This does not
+     * generalize.
+     */
+    signal(SIGHUP, SIG_IGN);
```

Comment:
If this is non-standard behavior AND iperf3 doesn't rely on it anyway because it ends up doing its own thing... then KISS.

Reply:
No, it's not non-standard at all, any more than any other daemon(3) behavior since it seems to vary between implementations.  The idea is to make sure the child isn't affected by a SIGHUP when the parent dies.  iperf3 installs a signal handler later so we don't need to worry about restoring the original value, but the effect is still there.  FreeBSD does this BTW.